tests/Http/ActionTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Tests\Http;
+
+use Tests\TestCase;
+use Rogue\Models\Action;
+use Rogue\Models\Campaign;
+
+class ActionTest extends TestCase
+{
+    /**
+     * Test that a POST request to /actions creates a new action.
+     *
+     * POST /actions
+     * @return void
+     */
+    public function testCreatingAnAction()
+    {
+        // Create a campaign that the action will belong to.
+        $campaign = factory(Campaign::class)->create();
+        $actionName = $this->faker->sentence;
+
+        // Create an action;
+        $this->actingAsAdmin()->postJson('actions', [
+            'name' => $actionName,
+            'campaign_id' => $campaign->id,
+            'post_type' => 'photo',
+            'reportback' => 1,
+            'civic_action' => 0,
+            'scholarship_entry' => 1,
+            'noun' => 'things',
+            'verb' => 'done',
+        ]);
+
+        // Make sure the action is persisted.
+        $this->assertDatabaseHas('actions', [
+            'name' => $actionName,
+            'campaign_id' => $campaign->id,
+        ]);
+    }
+
+    /**
+     * Test that a GET request to /api/v3/actions returns an index of all campaigns.
+     *
+     * GET /api/v3/actions
+     * @return void
+     */
+    public function testActionIndex()
+    {
+        // Create five actions.
+        factory(Action::class, 5)->create();
+
+        // Create a few more for a different campaign.
+        factory(Action::class, 3)->create();
+
+        // And one more for good measure.
+        factory(Action::class, 2)->create();
```

Comment:
@chloealee does each grouping get the same `campaign_id` or will they all be different? I don't remember exactly how this works

Reply:
ohhh @katiecrane you're right! So I changed it back because even though it creates 5 actions at the same time, each action has a different `campaign_id`

Reply:
ah right! because of the factory update! I love it!